### PR TITLE
chore(docs): Update the Basic Architecture page

### PR DESCRIPTION
**Pull request by anupriya0703**
The following updates have been made:
- The architectural components table has been updated.
- The description of some of these components(below the table) has been updated.

Signed-off-by: anupriya0703 <anupriya.gupta@mayadata.io>

### CODE REVIEW

File: reference/basic-architecture.md [hunk 1]
```diff
@@ -9,61 +9,62 @@ More detailed guides to Mayastor's components, their design and internal structu
...
-| *mayastor-etcd* | Service | Exposes NATS message bus end point | Single |
-| *mayastor-etcd-headless* | Service | Exposes NATS message bus end point | Single |
-| **Data Plane**|
-| mayastor| Pod | Hosts Mayastor I/O engine| User-selected nodes |
-| **k8s Resource Types**  |
-| mayastorpools | CRD | Declares a Mayastor pool's desired state and reflects its current state | User-defined, one or many |
+| agent-core | Deployment | Principle control plane actor | Single |
+| csi-controller | Deployment | Hosts Mayastor's CSI controller implementation and CSI provisioner side car| Single |
+| api-rest | Pod | Hosts the public API REST server | Single |
+| *api-rest*  | Service | Exposes the REST API server via NodePort |
```
Comment (avishnu) [inline reference/basic-architecture.md]: @tiagolobocastro is there a requirement to expose this on nodePortl?

File: reference/basic-architecture.md [hunk 2]
```diff
@@ -9,61 +9,62 @@ More detailed guides to Mayastor's components, their design and internal structu
...
+| operator-diskpool | Deployment | Hosts DiskPool operator | Single |
+| csi-node| DaemonSet | Hosts CSI Driver node plugin containers | All worker nodes |
+| etcd | StatefulSet | Hosts etcd Server container | Configurable(<i>Recommended: Three replicas)</i> |
+| etcd | Service | Exposes etcd DB endpoint | Single |
+| etcd-headless | Service | Exposes etcd DB endpoint | Single |
+| io-engine| DaemonSet | Hosts Mayastor I/O engine| User-selected nodes |
+| DiskPool | CRD | Declares a DiskPool's desired state and reflects its current state | User-defined, one or many |
```
Comment (avishnu) [inline reference/basic-architecture.md]: In the description, should we say Mayastor pool's desired state...

File: reference/basic-architecture.md [hunk 3]
```diff
@@ -9,61 +9,62 @@ More detailed guides to Mayastor's components, their design and internal structu
...
+| loki | Service| Exposes the Loki API endpoint via ClusterIP | Single |
 
 ## Component Roles
 
-### Control Plane
+### io-engine
 
-A microservices patterned control plane, centered around a core agent with a publically exposed RESTful API.  This is extended by a dedicated operator responsible for managing the life cycle of "Mayastor Pools" (an abstraction for devices supplying the cluster with persistent backing storage), and a CSI compliant external provisioner (controller).  The source for the control plane components is located in its [own repository](https://github.com/mayadata-io/mayastor-control-plane)
+The io-engine pods encapsulate Mayastor containers, which implement the I/O path from the block devices at the persistence layer, up to the relevant initiators on the worker nodes mounting volume claims.
+The instance of the Mayastor running inside the container performs four major classes of functions:
```
Comment (avishnu) [inline reference/basic-architecture.md]: Missing continuity between the sentence at line-34 and this.

File: reference/basic-architecture.md [hunk 4]
```diff
@@ -9,61 +9,62 @@ More detailed guides to Mayastor's components, their design and internal structu
...
+| loki | StatefulSet| Stores the historical logs exported by promtail pods | Single |
+| loki | Service| Exposes the Loki API endpoint via ClusterIP | Single |
 
 ## Component Roles
 
-### Control Plane
+### io-engine
 
-A microservices patterned control plane, centered around a core agent with a publically exposed RESTful API.  This is extended by a dedicated operator responsible for managing the life cycle of "Mayastor Pools" (an abstraction for devices supplying the cluster with persistent backing storage), and a CSI compliant external provisioner (controller).  The source for the control plane components is located in its [own repository](https://github.com/mayadata-io/mayastor-control-plane)
+The io-engine pods encapsulate Mayastor containers, which implement the I/O path from the block devices at the persistence layer, up to the relevant initiators on the worker nodes mounting volume claims.
```
Comment (avishnu) [inline reference/basic-architecture.md]: pod

File: reference/basic-architecture.md [hunk 5]
```diff
@@ -9,61 +9,62 @@ More detailed guides to Mayastor's components, their design and internal structu
...
+The instance of the Mayastor running inside the container performs four major classes of functions:
+- Presents a gRPC interface to the components which allow it to orchestrate the creation, configuration, and deletion of Mayastor-managed objects hosted by that instance.
```
Comment (avishnu) [inline reference/basic-architecture.md]: This can be the 4th point and be modified as "Provides a gRPC interface service to orchestrate the creation, deletion and management of the above objects, hosted on that node."

File: reference/basic-architecture.md [hunk 6]
```diff
@@ -9,61 +9,62 @@ More detailed guides to Mayastor's components, their design and internal structu
...
-### NATS
+### etcd
+   
+etcd is a distributed reliable key-value store for the critical data of a distributed system. Mayastor uses etcd as a reliable persistent store for its configuration and state data.
 
-[NATS](https://nats.io/) is a high performance open source messaging system. It is used by Mayastor as the transport mechanism for registration messages passed between Mayastor I/O engine pods running in the cluster and the core agent component, which maintains an inventory of active Mayastor nodes.  It is also used as the transport abstraction for inter data-plane / control-plane gRPC calls.
+### Supportability:
 
-### etcd
+The supportability tool is used to create support bundles (archive files) by interacting with multiple services present in the system. These bundles contain information about the entire Mayastor system, as well as specific Mayastor resources like volumes, pools and nodes, and can be used for debugging. It can collect the following information:
```
Comment (avishnu) [inline reference/basic-architecture.md]: "present in the cluster where Mayastor is installed".

The tool can collect the following information:

File: reference/basic-architecture.md [hunk 7]
```diff
@@ -9,61 +9,62 @@ More detailed guides to Mayastor's components, their design and internal structu
...
+- Topological information of Mayastor's resource(s) by interacting with the REST service
```
Comment (avishnu) [inline reference/basic-architecture.md:60]: Mayastor resources like (mention the list)

File: reference/basic-architecture.md [hunk 8]
```diff
@@ -9,61 +9,62 @@ More detailed guides to Mayastor's components, their design and internal structu
...
 
-"[etcd](https://github.com/etcd-io/etcd) is a distributed reliable key-value store for the most critical data of a distributed system"
+### Loki:
 
-Mayastor uses etcd as a reliable persistent store for its configuration and state data.
+    [Loki](https://grafana.com/oss/loki/) aggregates and centrally stores logs from all Mayastor containers which are deployed in the cluster.
 
+### Promtail:
+     
+    Promtail is a log collector built specifically for Loki. It uses the configuration file for target discovery and includes analogous features for labeling, transforming, and filtering logs from containers before ingesting them to Loki.
```
Comment (avishnu) [inline reference/basic-architecture.md]: For the sake of heirarchy, I think we should keep Loki and Promtail, and also kubectl supportability commands under the supportability section giving the notion of a supportability stack.

Comment (avishnu) [inline reference/basic-architecture.md]: The external link for Promtail can be provided.

File: reference/basic-architecture.md (hunk 1 above)
Comment (tiagolobocastro) [reply reference/basic-architecture.md]: No, we don't need to expose any ports now

File: reference/basic-architecture.md (hunk 8 above)
Comment (GlennBullingham) [reply reference/basic-architecture.md]: I think we should leave this small reference here - it's purpose to to just inform the user that this pod is an expected part of their deployment, and what it is doing on their cluster.  The actual "care and feeding" can indeed be part of the supportability section.

File: reference/basic-architecture.md [hunk 9]
```diff
@@ -9,61 +9,62 @@ More detailed guides to Mayastor's components, their design and internal structu
...
+| loki | Service| Exposes the Loki API endpoint via ClusterIP | Single |
 
 ## Component Roles
 
-### Control Plane
+### io-engine
 
-A microservices patterned control plane, centered around a core agent with a publically exposed RESTful API.  This is extended by a dedicated operator responsible for managing the life cycle of "Mayastor Pools" (an abstraction for devices supplying the cluster with persistent backing storage), and a CSI compliant external provisioner (controller).  The source for the control plane components is located in its [own repository](https://github.com/mayadata-io/mayastor-control-plane)
+The io-engine pod encapsulates Mayastor containers, which implement the I/O path from the block devices at the persistence layer, up to the relevant initiators on the worker nodes mounting volume claims.
+The instance of the Mayastor running inside the container performs four major classes of functions:
```
Comment (GlennBullingham) [inline reference/basic-architecture.md]:
```suggestion
The Mayastor process running inside this container performs four major functions:
```
(It's a bit wordy)

**Review by GlennBullingham — APPROVED** [review]

File: reference/basic-architecture.md (hunk 1 above)
Comment (anupriya0703) [reply reference/basic-architecture.md]: Removed `via NodePort`

File: reference/basic-architecture.md (hunk 8 above)
Comment (avishnu) [reply reference/basic-architecture.md]: In that case, I notice couple of inconsistencies here:
1. There is a "supportability" section above this "Promtail" section, and there is no tangible entity representing "supportability" as in a deployment/daemonset.
2. We don't have sections for other components here like api-server, core, pool-operator, etc.

Comment (anupriya0703) [reply reference/basic-architecture.md]: Based on our call earlier, I will be addressing these as part of another PR.

**Review by avishnu — APPROVED** [review]
LGTM

**Comment by anupriya0703** [comment]
This PR cannot be merged because of the failing commit linter. Raising a new PR ([PR 65](https://github.com/openebs/mayastor-docs/pull/65)).
